api-controllers/ServiceBrokerApiController.js

```diff
@@ -44,72 +49,167 @@ class ServiceBrokerApiController extends FabrikBaseController {
-        dashboard_url: req.instance.dashboardUrl
+        dashboard_url: getDashboardUrl(params.service_id, params.plan_id, req.params.instance_id)
       };
-      if (req.instance.async) {
+      if (plan.manager.async) {
         statusCode = CONST.HTTP_STATUS_CODE.ACCEPTED;
-        body.operation = utils.encodeBase64(result);
       }
       res.status(statusCode).send(body);
     }
 
-    function conflict(err) {
-      /* jshint unused:false */
+    function conflict() {
       res.status(CONST.HTTP_STATUS_CODE.CONFLICT).send({});
     }
 
     req.operation_type = CONST.OPERATION_TYPE.CREATE;
-
     return Promise
-      .try(() => req.instance.create(params))
+      .try(() => {
+        return eventmesh.apiServerClient.createResource({
+          resourceGroup: plan.manager.resource_mappings.resource_group,
+          resourceType: plan.manager.resource_mappings.resource_type,
+          resourceId: req.params.instance_id,
+          options: params,
+          status: {
+            state: CONST.APISERVER.RESOURCE_STATE.IN_QUEUE,
+            lastOperation: {},
+            response: {}
+          }
+        });
+      })
+      .then(() => {
+        if (!plan.manager.async) {
+          return eventmesh.apiServerClient.getResourceOperationStatus({
+            resourceGroup: plan.manager.resource_mappings.resource_group,
+            resourceType: plan.manager.resource_mappings.resource_type,
+            resourceId: req.params.instance_id,
+            start_state: CONST.APISERVER.RESOURCE_STATE.IN_QUEUE,
+            started_at: new Date()
+          });
+        }
+      })
       .then(done)
-      .catch(ServiceInstanceAlreadyExists, conflict);
+      .catch(Conflict, conflict);
   }
 
   patchInstance(req, res) {
     const params = _
       .chain(req.body)
-      .omit('plan_id', 'service_id')
+      // .omit('plan_id', 'service_id') //Not omiting even for restore. will that be a problem?
       .cloneDeep()
       .value();
     //cloning here so that the DirectorInstance.update does not unset the 'service-fabrik-operation' from original req.body object
+    const isRestoreOperation = _.get(params.parameters, 'service-fabrik-operation', null) ? true : false;
+    req.operation_type = CONST.OPERATION_TYPE.UPDATE;
+    const planId = params.plan_id;
+    const plan = catalog.getPlan(planId);
 
     function done(result) {
       let statusCode = CONST.HTTP_STATUS_CODE.OK;
       const body = {};
-      if (req.instance.async) {
+      if (plan.manager.async) {
         statusCode = CONST.HTTP_STATUS_CODE.ACCEPTED;
-        body.operation = utils.encodeBase64(result);
+        if (isRestoreOperation) {
+          body.operation = utils.encodeBase64(result);
+        }
       }
       res.status(statusCode).send(body);
     }
 
-    req.operation_type = CONST.OPERATION_TYPE.UPDATE;
+    function isUpdatePossible(previousPlanId) {
+      const planId = params.plan_id;
+      const plan = catalog.getPlan(planId);
```

Comment:
Removed the lines.

Reply:
Didn't fully understand the second comment. Even though plan id is not a mandatory parameter, we expect platforms to pass it as `assigninstance` and `assignmanager` would otherwise fail. This is also done keeping the same thought. Let us discuss if you see any issue.

Reply:
params.plan_id is not mandatory. If params.plan_id not present we should throw BadRequest. But we are throwing ServicePlanNotFound.